stdlib/out_channel.mli

```diff
@@ -0,0 +1,182 @@
+    output and standard error at the right time. *)
+
+val flush_all : unit -> unit
+(** Flush all open output channels; ignore errors. *)
+
+val output_char : t -> char -> unit
+(** Write the character on the given output channel. *)
+
+val output_string : t -> string -> unit
+(** Write the string on the given output channel. *)
+
+val output_bytes : t -> bytes -> unit
+(** Write the byte sequence on the given output channel. *)
+
+val output : t -> bytes -> int -> int -> unit
+(** [output oc buf pos len] writes [len] characters from byte sequence [buf],
+    starting at offset [pos], to the given output channel [oc].
+    Raise [Invalid_argument "output"] if [pos] and [len] do not
+    designate a valid range of [buf]. *)
+
+val output_substring : t -> string -> int -> int -> unit
+(** Same as [output] but take a string as argument instead of
+    a byte sequence. *)
+
+val output_byte : t -> int -> unit
+(** Write one 8-bit integer (as the single character with that code)
+    on the given output channel. The given integer is taken modulo
+    256. *)
+
+val output_binary_int : t -> int -> unit
+(** Write one integer in binary format (4 bytes, big-endian)
+    on the given output channel.
+    The given integer is taken modulo 2{^32}.
+    The only reliable way to read it back is through the
+    {!input_binary_int} function. The format is compatible across
+    all machines for a given version of OCaml. *)
+
+val seek : t -> int64 -> unit
+(** [seek chan pos] sets the current writing position to [pos]
+    for channel [chan]. This works only for regular files. On
+    files of other kinds (such as terminals, pipes and sockets),
+    the behavior is unspecified. *)
+
+val pos : t -> int64
+(** Return the current writing position for the given channel.  Does
+    not work on channels opened with the [Open_append] flag (returns
+    unspecified results). *)
+
+val length : t -> int64
+(** Return the size (number of characters) of the regular file
+    on which the given channel is opened.  If the channel is opened
+    on a file that is not a regular file, the result is meaningless. *)
+
+val set_binary_mode : t -> bool -> unit
+(** [set_binary_mode oc true] sets the channel [oc] to binary
+    mode: no translations take place during output.
+    [set_binary_mode oc false] sets the channel [oc] to text
+    mode: depending on the operating system, some translations
+    may take place during output.  For instance, under Windows,
+    end-of-lines will be translated from [\n] to [\r\n].
+    This function has no effect under operating systems that
+    do not distinguish between text mode and binary mode. *)
+
+val output_lines : t -> string list -> unit
+(** Outputs a list of lines, each terminated by a newline character *)
+
+(** {2 Output functions on filenames} *)
+
+val write_lines : string -> string list -> unit
+(** TBA *)
+
+val write_all : filename:string -> string -> unit
+(** TBA *)
```

Comment:
`write_all` inconsistently has the `filename` arg, which I guess makes sense since you want to distinguish the arguments, but I think if it's here, it should go in the analogous functions.

I personally think these 2 should still be `file_write_lines` and `file_write_all`. A little longer, but you're saving on opening the file, and it helps clarify their purpose.

If you keep the argument name, it could be shortened to `file` for convenience -- it's obvious that it's a name from the type.

Reply:
I'd rather have the label remain `filename` because it's explicit.  It can't be confused with `file_content` then.